src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,180 @@
+    if (!privateChats) return {};
+    return privateChats.reduce<LastMessageByUser>(lastMessageByUserReducer, {});
+  }, [lastMessageByUserReducer, privateChats]);
+
+  const onClickOnSender = useCallback(
+    (sender: WithId<User>) => {
+      const chatsToUpdate = privateChats.filter(
+        (chat) => !chat.isRead && chat.from === sender.id
+      );
+      chatsToUpdate.map(
+        (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+      );
+      setSelectedUser(sender);
+    },
+    [privateChats, user]
+  );
+
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [privateChats, selectedUser, HIDE_BEFORE]
+  );
+
+  const chatContext = useContext(ChatContext);
+  const onMessageSubmit = useCallback(
+    async (data: { messageToTheBand: string }) => {
+      chatContext &&
+        user &&
+        chatContext.sendPrivateChat(
+          user.uid,
+          selectedUser!.id,
+          data.messageToTheBand
+        );
+    },
+    [chatContext, selectedUser, user]
+  );
+
+  const hasPrivateChats = !isEmpty(discussionPartnerWithLastMessageExchanged);
+
+  const discussions = useMemo(() => {
+    return Object.keys(discussionPartnerWithLastMessageExchanged).sort((a, b) =>
+      discussionPartnerWithLastMessageExchanged[b].ts_utc
+        .valueOf()
+        .localeCompare(
+          discussionPartnerWithLastMessageExchanged[a].ts_utc.valueOf()
+        )
+    );
+  }, [discussionPartnerWithLastMessageExchanged]);
+
+  return selectedUser ? (
+    <>
+      <div
+        className="private-container-back-btn"
+        onClick={() => setSelectedUser(undefined)}
+      />
+      <ChatBox chats={chatsToDisplay} onMessageSubmit={onMessageSubmit} />
+    </>
+  ) : (
+    <>
+      {hasPrivateChats && (
+        <div className="private-container show">
+          <div className="private-messages-list">
+            {discussions.map((userId: string) => {
+              const sender = { ...chatUsers![userId], id: userId };
+              const lastMessageExchanged =
+                discussionPartnerWithLastMessageExchanged?.[userId];
+              const profileName = sender.anonMode
+                ? DEFAULT_PARTY_NAME
+                : sender.partyName;
+
+              return (
+                <div
+                  key={userId}
+                  className="private-message-item"
+                  onClick={() => onClickOnSender(sender)}
```

Comment:
This will still cause re-renders due to the wrapping function reference (`() =>`) changing on every render.

See previous comment RE: 'closing around' the state to avoid this.

Reply:
@denisdimitrov this one still needs to be done it seems

Reply:
`=\{\(\) =>` has hundreds of references throughout the codebase, is there a better solution than refactoring it all by hand?

Reply:
@0xdevalias this is not possible to refactor, because `sender` is inside the `map()` call so not available outside the inner function

Reply:
Then I believe react will see it as a new reference and have to re-render. I think it is less 'painful' if it's a 'leaf' element, eg. not much nested beneath it, but where we have deeply nested things that use it, I believe that will hurt more.

Reply:
Not sure if there is a better solution than manually. A linter could help you find them, but probably just as good as a find would. Often times a lot of places where they are used/passed down through components can be solved 'better' by making a more specific function higher up, and passing that instead.